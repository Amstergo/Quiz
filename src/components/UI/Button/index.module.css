.Button {
  display: inline-block;
  vertical-align: middle;
  transform: perspective(1px) translateZ(0);
  box-shadow: 0 0 1px rgba(0, 0, 0, 0);
  position: relative;
  background-color: transparent;
  color: white;
  border: none;
  cursor: pointer;
  margin: 15px 10px;
  text-decoration: underline;
  line-height: 1.7;
}

.Button::before {
  pointer-events: none;
  content: "";
  position: absolute;
  border: #e1e1e1c7 solid 2px;
  border-radius: 5px;
  top: -16px;
  right: -16px;
  bottom: -16px;
  left: -16px;
  opacity: 0;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: top, right, bottom, left;
  transition-property: top, right, bottom, left;
}

.Button:hover::before {
  top: -8px;
  right: -8px;
  bottom: -8px;
  left: -8px;
  opacity: 1;
}

.Button:disabled {
  cursor: not-allowed;
  background: #ccc;
}
