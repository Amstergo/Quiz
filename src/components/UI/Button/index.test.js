import React from "react";
import Button from "./index";
import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

configure({
  adapter: new Adapter()
});

describe("Render Button", () => {
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });

  it("Should render correctly button component", () => {
    const ButtonComponent = renderer.create(<Button />).toJSON();
    //const tree = shallow(<Button className="button test" />);
    expect(ButtonComponent).toMatchSnapshot();
  });

  it("check the onChange callback", () => {
    const onChange = jest.fn();
    // const props = {
    //   onChange
    // };
    //ButtonComponent = mount(<Button {...props} />);
    const tree = shallow(<Button onClick={onChange} />);
    tree.simulate("click");
    expect(onChange).toHaveBeenCalled();
  });
});
