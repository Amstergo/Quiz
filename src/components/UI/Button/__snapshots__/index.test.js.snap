// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Should render correctly button component 1`] = `
<button
  className="Button undefined"
/>
`;
